pepys_import/core/store/data_store.py

```diff
@@ -590,8 +589,43 @@ def search_table_type(self, table_type_id):
 
     #############################################################
     # New methods
+    def find_datafile(self, datafile_name):
+        """
+        This method tries to find a Datafile entity with the given datafile_name. If it
+        finds, it returns the entity. If it is not found, it searches synonyms.
 
-    def get_datafile(self, datafile_name, datafile_type):
+        :param datafile_name:  Name of Datafile
+        :type datafile_name: String
+        :return:
+        """
+        datafile = (
+            self.session.query(self.db_classes.Datafile)
+            .filter(self.db_classes.Datafile.reference == datafile_name)
+            .first()
+        )
+        if datafile:
+            return datafile
+
+        synonym = (
+            self.session.query(self.db_classes.Synonym)
+            .filter(
+                self.db_classes.Synonym.synonym == datafile_name,
+                self.db_classes.Synonym.table == "Datafiles",
+            )
+            .first()
+        )
+        if synonym:
+            datafile = (
+                self.session.query(self.db_classes.Datafile)
+                .filter(self.db_classes.Datafile.datafile_id == synonym.entity)
+                .first()
+            )
+            if datafile:
+                return datafile
+
+        return None
+
+    def get_datafile(self, datafile_name=None, datafile_type=None):
```

Comment:
@BarisSari - I think `get_datafile` could do with a method comment, particularly since we need to explain the difference between `get` and `find` methods for `_datafile`.

Reply:
I wrote comments for it. If it is still not clear, I can add more:

https://github.com/debrief/pepys-import/pull/73/commits/552af3a5ec0e9fb2860626e6168ececac7c3e5bd